src/sentry/utils/snuba.py

```diff
@@ -926,16 +928,33 @@ def resolve_snuba_aliases(snuba_filter, resolve_func, function_translations=None
                 name = resolve_func(col)
 
             groupby[idx] = name
-        resolved.groupby = groupby
 
     aggregations = resolved.aggregations
+    to_append = []
     for aggregation in aggregations or []:
         derived_columns.add(aggregation[2])
         if isinstance(aggregation[1], six.string_types):
-            aggregation[1] = resolve_func(aggregation[1])
+            if aggregation[1].startswith("metrics."):
+                found = False
+                for col in selected_columns:
+                    if col[2].strip("`") == aggregation[1]:
+                        found = True
+                        continue
+                if not found:
+                    aggregation[
+                        0
+                    ] += "(toFloat32OrNull(tags.value[indexOf(tags.key, '{}')]))".format(
```

Comment:
I know that this is experimental.
With that being said, anyone who steps has concerns about snuba being slow on requesting these columns will just get linked this comment.

Reply:
Sounds good. 

This entire PR is not expected to be merged into sentry.

Reply:
ssh, ssh, keep your voice down. it's a super duper secret. they'll hear you 👀

Reply:
wait, so I was not supposed to post this on reddit ?

Reply:
damn is it hackweek already